File: C++ 28tech T7_2024/contest_3/3.cpp
// tổng bội 3 
//nhập vào n không quá 10^6, tính tổng các số nguyên dương không vượt quá n và chia hết cho 3

#include <iostream>
using namespace std;

int main(){
	int n;cin >> n;
	long long sum = 0;
	for(int i = 1;i<=n;i++){
		if(i%3==0){
			sum +=i;
		}
	}
	cout << sum << endl;
}

File: Khác/hsg_2024/đề 3/3.py
# Bài 3. Để tính giá trị (số nguyên) gần đúng căn bậc hai của số tự nhiên n cho trước:, người ta đã thiết lập hàm sau với ý tưởng gần tương tự thuật toán tìm kiếm tuần tự như sau: 
 
# Hãy thiết kế lại thuật toán tìm số nguyên lớn nhất không vượt quá căn bậc hai của n bằng kĩ thuật chia để trị.

def n(n):
    if n < 0:
        return False
    if n == 0 or n == 1:
        return n
    
    low, high = 0, n
    result = 0

    while low <= high:
        mid = (low + high) // 2
        if mid * mid == n:
            return mid  
        elif mid * mid < n:
            result = mid  
            low = mid + 1
        else:
            high = mid - 1
    
    return result


print(n(100000000044233423845973498757843))